### PR TITLE
[FEATURE] Create commit hooks for build, test, checksum update and optimisation on commit #223

### LINKED ISSUE #223: [FEATURE] Create commit hooks for build, test, checksum update and optimisation on commit

### PR DESCRIPTION
**Pull request by deveusss**
closes #223

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -54,7 +54,7 @@ download-deps:
 	wget https://github.com/CosmWasm/cw-plus/releases/latest/download/cw20_base.wasm -O artifacts/cw20_base.wasm
 # TODO?: test dao-contracts
 
-all: build test check schema gen optimize checksum
+all: build test lint schema gen optimize checksum
```
Comment (Buckram123) [inline justfile:57]: Let's have lint before builds and tests!
```suggestion
all: lint build test schema gen optimize checksum
```

File: packages/cw-croncat-core/Cargo.toml [hunk 2]
```diff
@@ -32,4 +32,5 @@ serde-json-wasm = { version = "0.4.1", git = "https://github.com/CyberHoward/ser
 [dev-dependencies]
 cosmwasm-schema = { version = "1.0.0" }
 cw-multi-test = { version = "0.16.0", features = ["staking"] }
-cw20-base = { version = "0.13.4", features = ["library"] }
\ No newline at end of file
+cw20-base = { version = "0.13.4", features = ["library"] }
+rusty-hook = "0.11.2"
```
Comment (Buckram123) [inline packages/cw-croncat-core/Cargo.toml:36]: Let's have new lines at the end of the files)
```suggestion
rusty-hook = "0.11.2"

```

File: packages/cw-rules-core/Cargo.toml [hunk 3]
```diff
@@ -20,3 +20,4 @@ cw20 = { version = "0.13.4" }
 
 [dev-dependencies]
 cosmwasm-schema = { version = "1.0.0" }
+rusty-hook = "0.11.2"
```
Comment (Buckram123) [inline packages/cw-rules-core/Cargo.toml:23]:
```suggestion
rusty-hook = "0.11.2"

```

File: contracts/cw-rules/Cargo.toml [hunk 4]
```diff
@@ -69,3 +69,4 @@ cw4 = "0.16.0"
 cw4-group = "0.16.0"
 base64 = "0.13.0"
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
+rusty-hook = "0.11.2"
```
Comment (Buckram123) [inline contracts/cw-rules/Cargo.toml:72]:
```suggestion
rusty-hook = "0.11.2"

```

File: contracts/cw-croncat/Cargo.toml [hunk 5]
```diff
@@ -60,3 +60,4 @@ cwd-core = { version = "0.2.0", git = "https://github.com/DA0-DA0/dao-contracts"
 cwd-interface = { version = "0.2.0", git = "https://github.com/DA0-DA0/dao-contracts" }
 cwd-proposal-single = { version = "0.2.0", git = "https://github.com/DA0-DA0/dao-contracts" }
 cwd-voting = { version = "0.2.0", git = "https://github.com/DA0-DA0/dao-contracts.git" }
+rusty-hook = "0.11.2"
```
Comment (Buckram123) [inline contracts/cw-croncat/Cargo.toml:63]:
```suggestion
rusty-hook = "0.11.2"

```

File: Cargo.toml [hunk 6]
```diff
@@ -18,4 +18,4 @@ lto = true
 debug-assertions = false
 panic = 'abort'
 incremental = false
-overflow-checks = true
+overflow-checks = true
```
Comment (Buckram123) [inline Cargo.toml:21]:
```suggestion
overflow-checks = true

```

File: .rusty-hook.toml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[hooks]
+pre-commit = "just all"
```
Comment (Buckram123) [inline .rusty-hook.toml:2]: I often do few commits before pushing. Is that's enough for pre-commit?
```suggestion
pre-commit = "just lint test"
pre-push = "just all"
```

**Review by Buckram123 — CHANGES_REQUESTED** [review]
Nice to have this cool tool!

File: justfile (hunk 1 above)
Comment (deveusss) [reply justfile:57]: this one is fixed in one of PRs))

**Review by Buckram123 — APPROVED** [review]

File: justfile [hunk 8]
```diff
@@ -1,6 +1,6 @@
 test_addrs := env_var_or_default('TEST_ADDR', `jq -r '.[].address' ci/test_accounts.json | tr '\n' ' '`)
 set export
-check:
+lint:
```
Comment (mikedotexe) [inline justfile:3]: 👍🏼

**Review by mikedotexe — APPROVED** [review]

**Pull request merged**